spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/ReactiveXsuaaJwtDecoder.java

```diff
@@ -0,0 +1,87 @@
+package com.sap.cloud.security.xsuaa.token.authentication;
+
+import java.text.ParseException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.jwt.JwtTimestampValidator;
+import org.springframework.security.oauth2.jwt.NimbusReactiveJwtDecoder;
+import org.springframework.security.oauth2.jwt.ReactiveJwtDecoder;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+import com.sap.cloud.security.xsuaa.XsuaaServiceConfiguration;
+
+import net.minidev.json.JSONObject;
+import reactor.core.publisher.Mono;
+
+public class ReactiveXsuaaJwtDecoder implements ReactiveJwtDecoder {
+
+	Cache<String, ReactiveJwtDecoder> cache;
+	private final XsuaaServiceConfiguration xsuaaServiceConfiguration;
+	private List<OAuth2TokenValidator<Jwt>> tokenValidators = new ArrayList<>();
+	private Collection<PostValidationAction> postValidationActions;
+
+	private static final String EXT_ATTR = "ext_attr";
+	private static final String ZDN = "zdn";
+	private static final String ZID = "zid";
+
+	// var arg it is only being converted to a List<OAuth2TokenValidator<Jwt>>,
+	// therefore its type safe.
+	ReactiveXsuaaJwtDecoder(XsuaaServiceConfiguration xsuaaServiceConfiguration, int cacheValidity, int cacheSize,
+			OAuth2TokenValidator<Jwt> tokenValidators, Collection<PostValidationAction> postValidationActions) {
+		cache = Caffeine.newBuilder().expireAfterWrite(5, TimeUnit.SECONDS).maximumSize(cacheSize).build();
+		this.xsuaaServiceConfiguration = xsuaaServiceConfiguration;
+
+		this.tokenValidators.add(new JwtTimestampValidator());
+		if (tokenValidators == null) {
+			this.tokenValidators.add(new XsuaaAudienceValidator(xsuaaServiceConfiguration));
+		} else {
+			this.tokenValidators.addAll(Arrays.asList(tokenValidators));
```

Comment:
Always a `DelegatingOAuth2TokenValidator` is passed, so you can simply add this:

```suggestion
			this.tokenValidators.add(tokenValidators);
```

But to me it also doesn't make sense to have this as a `List` internally.

Reply:
Done  with #https://github.com/SAP/cloud-security-xsuaa-integration/pull/103/commits/6a792bfbd752052a09be139248b0c1188ccdfe5c